src/bundle/Controller/ContentViewController.php

```diff
@@ -59,6 +60,31 @@ public function detailsTabAction(ContentView $view)
         return $view;
     }
 
+    public function locationsTabAction(ContentView $view)
+    {
+        $versionInfo = $view->getContent()->getVersionInfo();
+        $contentInfo = $versionInfo->getContentInfo();
+
+        if ($contentInfo->published) {
+            $locationRepository = $this->getRepository()->getLocationService();
+            $locations = $locationRepository->loadLocations($contentInfo);
+
+            $locations = array_map(function (Location $location) {
+                return new LocationDecorator($location);
+            }, $locations);
+
+            foreach ($locations as $location) {
+                $location->childCount = $locationRepository->getLocationChildCount($location->getValueObject());
```

Comment:
`childCount` can be set in the callback above, after creating the `LocationDecorator` object.